### ISSUE TITLE
OS X 10.11 kernel communication tests are hanging

### ISSUE THREAD

**Issue by theopolis** [facebook/osquery#2115]
When running the OS X kernel tests:
```
Total Test time (real) =  87.28 sec
cd build/darwin && cmake ../../ && \
		CTEST_OUTPUT_ON_FAILURE=1 make --no-print-directory kernel-test/fast
-- Building for platform OS X (darwin, 10.11)
-- Building osquery version  1.7.4-29-g10664a4 sdk 1.7.4
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/reed/git/github/osquery/build/darwin

Running kernel tests requires root.
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from KernelCommunicationTests
[ RUN      ] KernelCommunicationTests.test_communication
```

This hangs, but is interruptible.

**Comment by theopolis** [comment]
The non-debug version:
```
* thread #1: tid = 0x22188d, 0x00007fff979b4872 libsystem_kernel.dylib`__ioctl + 10, queue = 'com.apple.main-thread', stop reason = signal SIGSTOP
  * frame #0: 0x00007fff979b4872 libsystem_kernel.dylib`__ioctl + 10
    frame #1: 0x00007fff979b1e27 libsystem_kernel.dylib`ioctl + 159
    frame #2: 0x000000010e321e1d osquery_kernel_tests`osquery::CQueue::kernelSync(int) + 63
    frame #3: 0x000000010e1c8d00 osquery_kernel_tests`osquery::KernelCommunicationTests_test_communication_Test::TestBody() + 360
    frame #4: 0x000000010e4529c6 osquery_kernel_tests`void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) + 95
    frame #5: 0x000000010e440e67 osquery_kernel_tests`testing::Test::Run() + 139
    frame #6: 0x000000010e441938 osquery_kernel_tests`testing::TestInfo::Run() + 208
    frame #7: 0x000000010e441f9b osquery_kernel_tests`testing::TestCase::Run() + 221
    frame #8: 0x000000010e448053 osquery_kernel_tests`testing::internal::UnitTestImpl::RunAllTests() + 675
    frame #9: 0x000000010e4532a0 osquery_kernel_tests`bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) + 93
    frame #10: 0x000000010e447d63 osquery_kernel_tests`testing::UnitTest::Run() + 121
    frame #11: 0x000000010e1c80bc osquery_kernel_tests`main + 362
    frame #12: 0x00007fff872895ad libdyld.dylib`start + 1
    frame #13: 0x00007fff872895ad libdyld.dylib`start + 1
```

**Comment by theopolis** [comment]
Debug view:
```
* thread #1: tid = 0x22c98d, 0x00007fff979b4872 libsystem_kernel.dylib`__ioctl + 10, queue = 'com.apple.main-thread', stop reason = signal SIGSTOP
  * frame #0: 0x00007fff979b4872 libsystem_kernel.dylib`__ioctl + 10
    frame #1: 0x00007fff979b1e27 libsystem_kernel.dylib`ioctl + 159
    frame #2: 0x000000010c4db5cf osquery_kernel_tests`osquery::CQueue::kernelSync(this=0x00007fff53b8e160, options=1) + 111 at circular_queue_user.cpp:87
    frame #3: 0x000000010c084ed1 osquery_kernel_tests`osquery::KernelCommunicationTests_test_communication_Test::TestBody(this=0x00007f8580f0b0f0) + 561 at kernel_communication_tests.cpp:75
    frame #4: 0x000000010c9ee568 osquery_kernel_tests`void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(object=0x00007f8580f0b0f0, method=21 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00, location="the test body")(), char const*) + 136 at gtest.cc:2078
    frame #5: 0x000000010c9d9c95 osquery_kernel_tests`void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(object=0x00007f8580f0b0f0, method=21 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00, location="the test body")(), char const*) + 133 at gtest.cc:2114
    frame #6: 0x000000010c9b0133 osquery_kernel_tests`testing::Test::Run(this=0x00007f8580f0b0f0) + 211 at gtest.cc:2150
    frame #7: 0x000000010c9b1656 osquery_kernel_tests`testing::TestInfo::Run(this=0x00007f8580f07250) + 230 at gtest.cc:2326
    frame #8: 0x000000010c9b2705 osquery_kernel_tests`testing::TestCase::Run(this=0x00007f8580f07680) + 245 at gtest.cc:2444
    frame #9: 0x000000010c9c022a osquery_kernel_tests`testing::internal::UnitTestImpl::RunAllTests(this=0x00007f8580f07380) + 922 at gtest.cc:4315
    frame #10: 0x000000010c9eff78 osquery_kernel_tests`bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(object=0x00007f8580f07380, method=90 fe 9b 0c 01 00 00 00 00 00 00 00 00 00 00 00, location="auxiliary test code (environments or event listeners)")(), char const*) + 136 at gtest.cc:2078
    frame #11: 0x000000010c9dc955 osquery_kernel_tests`bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(object=0x00007f8580f07380, method=90 fe 9b 0c 01 00 00 00 00 00 00 00 00 00 00 00, location="auxiliary test code (environments or event listeners)")(), char const*) + 133 at gtest.cc:2114
    frame #12: 0x000000010c9bfdf6 osquery_kernel_tests`testing::UnitTest::Run(this=0x000000010ce3d9e0) + 422 at gtest.cc:3926
    frame #13: 0x000000010c07a143 osquery_kernel_tests`RUN_ALL_TESTS() + 35 at gtest.h:2288
    frame #14: 0x000000010c077b0b osquery_kernel_tests`main(argc=1, argv=0x00007fff53b8eb60) + 875 at tests.cpp:131
    frame #15: 0x00007fff872895ad libdyld.dylib`start + 1
```

Frame (2), the relevant one:
```
(osquery::CQueue *) this = 0x00007fff53b8e160
(int) options = 1
(osquery_buf_sync_args_t) sync = (options = 1, read_offset = 2726924, max_read_offset = 2726924, drops = 0)
(int) err = 0
(uint8_t *) new_max_read = 0x000000010dc01c0c "HHHHHHHHHHHHHHHHHHHH\x04"
```

This may be a regression introduced by me, switching the userland queue to be non-blocking.

**Comment by theopolis** [comment]
Fixed with #2116.
